### PR TITLE
Add Fee Reserve Feature

### PR DESCRIPTION
**Pull request by TingALin**
Edited by Ross:

Adds a feature to add a reserve for transaction fees, that exists on top of the existential deposit. `feeReserve` affects bonding and fast unstake deposits only (anything that requires locking or sending funds from your account). This ensures that existing funds will remain to pay for transaction fees.

Along with a slider that reflects a configurable total txFee value, the actual reserved value is reflected in `BalanceChart`.

Screenshot 2023-06-08 at 16 49 11

Screenshot 2023-06-08 at 16 49 05

On the above Update Reserve button:
- 2 check marks: existential deposit reserve and fee reserve are set.
- 1 check mark: one of the deposits have been set.
- 0 check marks: no funds are being reserved.

### CODE REVIEW

**Comment by rossbulat** [comment]
There is still work to be done here.

1. The button styles in the footer are different - why did you opt for the "Update Reserve" button to be a primary smaller button than the one next to it? This is basic design consistency.

2. There is no UI telling me that the reserve amount has been configured. We discussed before having a badge in the button that tells you the amount of the reserve you have configured. Nothing like this has been done.

3. There is something wrong with the logic. I cannot submit non-bonding transactions. It is telling me I do not have enough DOT. This was not tested - I expect big changes like this to be thoroughly tested.

There are more issues that I will leave as comments.

File: src/modals/UpdateReserve/index.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                  }
+                }}
+                {...sliderProps}
+              />
+            </div>
+          </div>
+          <p>{t('reserveText')}</p>
+          <div className="confirm">
+            <ButtonPrimaryInvert
+              text={t('confirm')}
```
Comment (rossbulat) [inline src/modals/UpdateReserve/index.tsx:139]: Change to `Done`. Confirm suggests that a change is being made when you click the button.

Also add a `Reset` button somewhere to set the reserve amount to zero.

File: src/modals/UpdateReserve/index.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        {warnings.length ? (
+          <ModalWarnings>
+            {warnings.map((warning, index) => (
+              <React.Fragment key={`warning_${index}`}>
+                {warning}
+              </React.Fragment>
+            ))}
+          </ModalWarnings>
+        ) : null}
+        <ActionItem
```
Comment (rossbulat) [inline src/modals/UpdateReserve/index.tsx:102]: Why are you duplicating titles? This is the same as the modal title.

File: src/modals/UpdateReserve/index.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    ? locksStaking.amount
+    : new BigNumber(0);
+  // available balance data
+  const fundsFree = planckToUnit(
+    allTransferOptions.freeBalance.minus(frozen.minus(lockStakingAmount)),
+    units
+  );
+
+  const warnings = [];
+  if (fundsFree.isZero()) {
```
Comment (rossbulat) [inline src/modals/UpdateReserve/index.tsx:53]: As discussed, we do not want this warning popping in as you are toggling the slider. Remove or reformat.

File: src/modals/UpdateReserve/index.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    warnings.push(<Warning text={t('reserveLimit')} />);
```
Comment (rossbulat) [inline src/modals/UpdateReserve/index.tsx:54]: This text makes little sense and has no context - not understandable. If you are struggling to write English text reach out to myself to add text content.

File: src/modals/UpdateReserve/index.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                    setReserve(
+                      new BigNumber(unitToPlanck(val.toString(), units))
+                    );
+                  }
+                }}
+                {...sliderProps}
+              />
+            </div>
+          </div>
+          <p>{t('reserveText')}</p>
```
Comment (rossbulat) [inline src/modals/UpdateReserve/index.tsx:136]: This text is not understandable.

File: src/modals/UpdateReserve/index.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+          inlineButton={
+            <ButtonHelp onClick={() => openHelp('Reserve Balance')} />
+          }
+        />
+
+        <WithSliderWrapper>
+          <h5 className="neutral">
+            {t('reserve')}
+            {ReserveFeedback && (
+              <span className="neutral">{ReserveFeedback.text}</span>
```
Comment (rossbulat) [inline src/modals/UpdateReserve/index.tsx:113]: This unit does not have `DOT` next to it. It is not obvious that it is token.

File: src/modals/UpdateReserve/index.tsx (hunk 3 above)
Comment (TingALin) [reply src/modals/UpdateReserve/index.tsx:53]: This pop-up is to remind the user they are reserving more than they have, so far this is the only way i can come up with, please let me know how i can do this reminding in a better way.

File: src/modals/UpdateReserve/index.tsx (hunk 4 above)
Comment (TingALin) [reply src/modals/UpdateReserve/index.tsx:54]: Please help to write the context in English if it is possible

File: src/modals/UpdateReserve/index.tsx (hunk 5 above)
Comment (TingALin) [reply src/modals/UpdateReserve/index.tsx:136]: Please help to write the context in English if it is possible.

**Comment by TingALin** [comment]
> There are more issues that I will leave as comments.

I misunderstood you and now the logic have changed. Other than the English text, all fixed now

**Pull request merged**